### ISSUE TITLE
size_t wraparound causes huge allocation when reading generic json  (again)

### ISSUE THREAD

**Issue by pauldreik** [stephenberry/glaze#1209]
Now when the generic json fuzzer can run after the previous problem was fixed,  I found a new problem in c7e4fbcb0c565442f49f57aa13dbc26e543e822c.

The following test fails with an exception from std::string:
```cpp
   "maybe not garbage"_test = []() {
      glz::json_t json{};
      std::string data = "\"\\uDBDD\" DDDD";
      auto r = glz::read_json(json, data);
      expect(r);
   };
```

It tries to allocate 18446744073709551614 bytes here:
https://github.com/stephenberry/glaze/blob/88bb45a12809ddf1a0a1e16d5172b0200211b24b/include/glaze/json/read.hpp#L776

**Comment by stephenberry** [comment]
Thanks so much for all these reports. I found the problem and fixed it in #1211.

I hadn't considered how null values would be valid hex digits, and thus we must check against the end iterator.

**Comment by pauldreik** [comment]
Unfortunately this is not dead yet as of 56a49eb607fa76a53b75c4bdba35453ff8d35d1d! This slightly modified test triggers:
```cpp
template <std::size_t N>
std::vector<char> make_vector(const char (&literal)[N])
{
   // include the null
   return {literal, literal + N + 1};
};
void test(){
   "maybe not garbage"_test = []() {
      glz::json_t json{};
      // std::string data = "\"\\uDBDD\" DDDD";
      auto data = make_vector("\"\\uDBDD\" DDDD");

      auto r = glz::read_json(json, data);
      expect(r);
   };
}

```

**Comment by stephenberry** [comment]
@pauldreik, this is actually an error in `make_vector`. The length of the char literal `N` already contains the null pointer, so adding `+ 1` reads beyond the length of the literal. You can just remove the `+ 1` and you'll have a null terminated `std::vector<char>`

**Comment by pauldreik** [comment]
sorry, my mistake. I must have pasted the wrong data. With the N+1 mistake corrected, the following causes an exception to be thrown:
      
```cpp
template <std::size_t N>
std::vector<char> make_vector(const char (&literal)[N])
{
   // include the null
   return {literal, literal + N};
};
void test(){
   "maybe not garbage"_test = []() {
      glz::json_t json{};
      auto data = make_vector("\"\\udb0f \"df33 ");
      auto r = glz::read_json(json, data);
      expect(r);
   };
}

```

**Comment by stephenberry** [comment]
Ah, thanks for this! There was a missing check for `\u` in the unicode surrogate pair parsing logic. Fixed in #1214
